src/cargo/util/toml/mod.rs

```diff
@@ -875,25 +1127,26 @@ impl TomlManifest {
                 }
             }
         }
-        if let Some(license_file) = &package.license_file {
+
+        if let Some(license_file) = ws_fields.license_file {
             let license_path = Path::new(&license_file);
             let abs_license_path = paths::normalize_path(&package_root.join(license_path));
             if abs_license_path.strip_prefix(package_root).is_err() {
                 // This path points outside of the package root. `cargo package`
                 // will copy it into the root, so adjust the path to this location.
-                package.license_file = Some(
+                package.license_file = Some(MaybeWorkspace::Defined(
                     license_path
                         .file_name()
                         .unwrap()
                         .to_str()
                         .unwrap()
                         .to_string(),
-                );
+                ));
             }
         }
         let all = |_d: &TomlDependency| true;
         return Ok(TomlManifest {
-            package: Some(package),
+            package: Some(Box::new(package)),
```

Comment:
Still don't understand why I needed to add this :thinking:

Reply:
I think its because you added this `package` function

https://github.com/rust-lang/cargo/pull/8664/files#diff-b7ba254e80a06d5a8f1bfc2fa3682383R994

The return type you placed in the function signature ends up making it do a deref coersion in order to convert the `&Box<TomlProject>` to a `&TomlProject`. If you change the return type of `package` to `CargoResult<&Box<TomlProject>>` it works out.

Reply:
O, of course, that makes perfect sense!